polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisResolvedPathWrapper.java

```diff
@@ -19,62 +19,41 @@
 package org.apache.polaris.core.persistence;
 
 import java.util.List;
-import java.util.stream.Collectors;
 import org.apache.polaris.core.entity.PolarisEntity;
 
 /**
  * Holds fully-resolved path of PolarisEntities representing the targetEntity with all its grants
  * and grant records.
  */
 public class PolarisResolvedPathWrapper {
-  private final List<ResolvedPolarisEntity> resolvedPath;
+  private final List<PolarisEntity> resolvedPath;
```

Comment:
Using plain `PolarisEntity` inside a "Resolved" entity path _wrapper_ does not sound good to me purely from a code readability point of view.

I'm fine using `PolarisEntity` in Authrorizer inputs, but perhaps we should rename the wrapper class and change its javadoc? What does "resolved" mean in this context?

If we merely want a dedicate class to hold a path of nested entities, how about `PolarisEntityPath`?

Reply:
This sounds fine to me - the main thing for me is to remove the different entity types so that each interface is dealing directly with the entity types (albeit, with a convenience wrapper in this case).